File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.py
class Solution:
    def maxArea(self, h: int, w: int, hcut: List[int], vcut: List[int]) -> int:

        mod = 10**9 + 7
        hcut.insert(0,0)
        hcut.append(h) 

        vcut.insert(0,0)
        vcut.append(w) 
    
        hcut.sort() ; vcut.sort() 
        maxh = maxv = 0 
        for i in range(1,len(hcut)):
                 maxh = max ( maxh ,( hcut[i]- hcut[i-1] ))

        for i in range(1, len(vcut)): 
            maxv = max(maxv , ( vcut[i] - vcut[i-1] )) 

        return maxv*maxh %(mod)

File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.cpp
class Solution {
public:
    int maxArea(int h, int w, vector<int>& hcut, vector<int>& vcut) {
        
        vcut.insert( vcut.begin() , 0) ;
        vcut.push_back(w) ; 
        hcut.insert(hcut.begin() , 0) ; 
        hcut.push_back(h) ; 
        sort(vcut.begin() , vcut.end()) ; 
        sort(hcut.begin() , hcut.end()) ; 
        
        int maxv = 0 , maxh = 0 ; 
        for(auto i=1 ; i<vcut.size() ; i++){
            maxv = max(maxv , vcut[i]-vcut[i-1]) ;
        }
        for(auto j=1 ; j < hcut.size() ; j++){
            maxh = max(maxh , hcut[j]-hcut[j-1]) ; 
        }
        
        return (long)maxh*maxv % 1000000007 ;  
    }
};